subprojects/core-api/src/main/java/org/gradle/api/artifacts/ModuleDependency.java

```diff
@@ -81,6 +86,14 @@ public interface ModuleDependency extends Dependency, HasConfigurableAttributes<
 
     /**
      * Returns the artifacts belonging to this dependency.
+     * <p>
+     * There will typically be a single artifact for a dependency, but this is not always true.  Modules can use
```

Comment:
> There will typically be a single artifact for a dependency

This wording here makes it seen each `ModuleDependency` will usually return a single artifact from this method. By default, `getArtifacts` will return an empty set unless the user adds an artifact themselves. 

I think we should revise this wording to be more clear here

Reply:
I'm all about adding more details to increase clarity.  I'll add that.